### ISSUE TITLE
E3D REVO Thermal Model

### ISSUE THREAD

**Issue by 3d-gussner** [prusa3d/Prusa-Firmware#4451]
1. Verify firmware version installed on your printer , to check this on the printers start screen the firmware version will be indicated you can also check which firmware version you already have via the following process L**CD menu -> Support. Scroll down** and you will see the firmware version.

If your printer displays a firmware version prior to 3.13.0 or it does not start with **E3DREVO_fw_** or **E3DREVO_HF_60W_** we would advise that you install the latest Revo supported firmware release via the following link and proceed with the next steps: https://github.com/prusa3d/Prusa-Firmware/releases
     
If you are using a E3D Revo firmware, please check that this matches the Revo Hotend you have installed as there are different versions.
- For a Standard 40W Revo heater core please use the. HEX file starting with **E3DREVO_fw_** ensuring you chose the correct. HEX file for the printer model you have.
- For a High flow 60W HeaterCore please use the. HEX file starting with **E3DREVO_HF_60W_** ensuring you chose the correct. HEX file for the printer model you have.

**If you do have an E3D Revo Firmware installed and it matches the Revo HotEnd you have please go to step 4.**

2. Once you have downloaded the firmware this Zip file and extracted it , please use the applicable. HEX file for both your printer model and the version of HeaterCore you have.
   - For a Standard 40W Revo heater core please use the. HEX file starting with **E3DREVO_fw_** ensuring you chose the correct. HEX file for the printer model you have.
   - For a High flow 60W HeaterCore please use the. HEX file starting with **E3DREVO_HF_60W_** ensuring you chose the correct. HEX file for the printer model you have.

_If you are unsure as to how to update your printers firmware with the .HEX file please refer to the Flashing procedure within the following guide:_ https://help.prusa3d.com/article/firmware-updating-mk3s-mk3s-mk3_2227

3. Once you have flashed and updated your printers firmware Run both the PID and TM calibration from the LCD Calibration menus of your machine.
4. If step 2 fails however please Use the Revo_Reset G-code files to force setting changes. There are 2 versions of this g-code that will need to be used based on the HeaterCore you have installed.
    - **[reset_revo.gcode.txt](https://github.com/prusa3d/Prusa-Firmware/files/12917795/reset_revo.gcode.txt)**: for a 40W HeaterCore. Please rename the file form `reset_revo.gcode.txt` to `reset_revo.gcode` 
    - **[reset_revo_hf_60W.gcode.txt](https://github.com/prusa3d/Prusa-Firmware/files/12917816/reset_revo_hf_60W.gcode.txt)**:  for a 60W High Flow HeaterCore. Please rename the file form `reset_revo_hf_60W.gcode.txt` to `reset_revo_hf_60W.gcode` 

These files will need to be copied to an SD card and installed  , to Install the G-code select it from the print from the SD menu on your printer and follow the on-screen instructions.

5. Once the G-code has been installed re-run both a PID and TM calibration from the LCD Calibration menus of your machine.

**Optional step:** If the above steps fail you may wish to factory reset the machine and run the setup wizard. Please note that as the process suggests this will reset the printer and its firmware to a factory state and any setting will be lost.

[Link to E3D ticket](https://e3d-online.zendesk.com/hc/en-us/requests/new?ticket_form_id=11304553253917)

**Comment by acothebraco** [comment]
Hello,

I have the same issute that my MK3S+ beeps three times with Thermal Anomaly and at the next time it stops with Thermal Error. I use the Revo Six Hotend and have never a problem with THERMAL ANOMALY. But since I put the MK3S+ in the Original Prusa Enclosure it shows up after 1-2 hours printing. I did also PID and Thermal Cal. on the 3.13.2 and the older 3.13.0 and I also tired to do a Revo_Reset G-Code without scuccess.  Only when I put the Printer outside of the Enclosure the Error will not apear. 

![IMG_7431](https://github.com/prusa3d/Prusa-Firmware/assets/80178563/9e15bfd6-44ee-47e3-9c74-004e8549b76e)

**Comment by Lino77** [comment]
same problem for me. I updated from 3.13.0 to 3.13.2, set the printer to factory settings, I can't even do the first layer of calibration. without the printer screaming, revo six for mk3

**Comment by Ezrae** [comment]
Adding my voice to this. I went through each step carefully. I reset TM and PID multiple times. I ran the gcode. And when I finally decided to factory reset, I was unable to finish the wizard due to the thermal errors I was continuing to get.

I've downgraded back to 3.11, and I'll try again with the next REVO firmware update.

**Comment by nemith** [comment]
Two printers, did all the steps here. All had the same issue.  One seems to actually finish the model but still fails later, the other can't even calibrate the model.  

Going to downgrade back to 3.11 this feature seems half baked and not suited for revo hotends.

**Comment by Lino77** [comment]
why downgrade , you can disable it.
[The thermal model protection can be disabled by connecting the printer to your computer (Octoprint, PuTTY...), and sending M310 S0. An additional M500 will save the setting and the Thermal model protection will remain disabled after the next restart.](https://help.prusa3d.com/article/thermal-model-calibration_382488#:~:text=model%20debug%20logging-,Disabling%20the%20Thermal%20model%20protection,the%20Thermal%20model%20protection%20will%20remain%20disabled%20after%20the%20next%20restart.,-Was%20this%20article)

**Comment by nemith** [comment]
> why , you can disable it.

I am selling this printer and it need to "just work".  Disabling until next restart isn't an option.

**Comment by Lino77** [comment]
M500 safe it and it will remain disable afer restart

**Comment by LoakAudio** [comment]
@3d-gussner , In the step 4 of your first post, isn't it "if step 3 fails" rather than "step 2" ? because it appends to me : changing the firmware went fine, reboot OK with the new firmware but PID calibration fails and TM calibration fails too. The gcode saved me.

**Comment by 3d-gussner** [comment]
@LoakAudio Thanks for the feedback I will check again.
Glad that the new firmware and gcode files saved you. :tada:

**Comment by bethdehart** [comment]
I'm in the same boat as others here. I have opened a ticket with E3D and will post back with any findings that might help others.

**Update** - I still have not heard back from E3D but I checked the **Thermistor extension cable** and it seems the wire has come lose from the housing. [Here](https://e3d-online.zendesk.com/hc/en-us/articles/4408428411153-Revo-Six-Assembly-Guide) are the docs if anyone is wondering.

**Comment by bethdehart** [comment]
@3d-gussner I happen to do a factory reset and after the calibration this happened

https://github.com/prusa3d/Prusa-Firmware/assets/6137749/24a2215f-4de8-4e8a-a89c-a9ace7ca4f83

Should I open a different Github issue? Please advise.

**Comment by 3d-gussner** [comment]
@jamesdehart Which calibration caused this screen?

**Comment by Skilbo** [comment]
After much head scratching, finally got it working, used the E3DREVO_fw version and everything worked first time. Thanks to @3d-gussner for laying out the procedures very much appreciated.

**Comment by bethdehart** [comment]
> @jamesdehart Which calibration caused this screen?

So it turned out my entire issue was because the thermal cable connector from the Hot-end to the cable that goes to the motherboard failed. The image shows the failed cable & connecor. One of the two wires were connected.

<img width="498" alt="image" src="https://github.com/prusa3d/Prusa-Firmware/assets/6137749/af4990f2-6710-4d6f-ace6-f98154d4e5a9">

For others who are running in to issues related to this hotend. It would be good to check there cable to see if there are any issues like I had. When contacting E3D it took them about a week to respond to the support tickets. They are really backed up.

I hope this helps.

**Comment by 3d-gussner** [comment]
@jamesdehart Thanks for sharing! This is an example why we implemented the Thermal model to detect issues like this and other.
Well done to investigate :medal_sports: instead of disabling the TM!

Side note: Try to add some more strain relief so all motions of the cable loom are possible without "pulling" cables at the connectors.

**Comment by acothebraco** [comment]
Thanos for sharing. I checked my printer six times and I don’t have any cable or hardware issues.  The point is that like I wrote in a other post that TM have problems when the printer is in a enclosure. 

Also I’d explained in this post how the problem occur in my setup https://github.com/prusa3d/Prusa-Firmware/issues/4431#issuecomment-1807171123

**Comment by ndemarco** [comment]
I'm chasing down the same `THERMAL ANOMALY` trouble,

- Verified the wiring is correct and in working order
- run [reset_revo.gcode.txt](https://github.com/prusa3d/Prusa-Firmware/files/12917795/reset_revo.gcode.txt)
- recalibrated PID
- recalibrated TA
- hard reset, followed by the usual recalibrations from the wizard

The PID and TA calibrations finished properly. Upon the next PLA preheat, `BEEP-BEEP THERMAL ANOMALY`.

**Comment by 3d-gussner** [comment]
@ndemarco Thanks for the update.
Is your printer in an enclosure?
Can you please try to capture the print in a logfile? Do you have a host system like Octoprint running?

**Comment by stevenbennett** [comment]
I'm also experiencing the issue described by users in this thread. Installed the Revo variant of 3.13.2 on two MK3S+ and now encounter consistent thermal anomaly errors. Things I've tried:

- verified heater/thermister cable and cable extension integrity
- ran reset_revo gcode mentioned in https://github.com/prusa3d/Prusa-Firmware/issues/4451#issue-1945318827
- performed full factory reset
- re-ran PID calibration
- re-ran thermal model calibration

After these steps I still encounter frequent thermal anomaly errors. Sometimes these occur right after initiating a preheat, sometimes right before a print begins, sometimes during a print. (I'm not using an enclosure). Going to try rolling back to 3.11 as other have suggested.

**Comment by slaesh** [comment]
had the same issue. it works now, maybe change the order of your steps 2 and 3 @stevenbennett :)
1. update firmware
2. factory reset
3. ran the gcode magic
4. PID calibration
5. thermal calibration

seems to work, step 3 was the magic sauce for me though! ;)

**Comment by 3d-gussner** [comment]
> had the same issue. it works now, maybe change the order of your steps 2 and 3 @stevenbennett :)
> 
>     1. update firmware
> 
>     2. factory reset
> 
>     3. ran the gcode magic
> 
>     4. PID calibration
> 
>     5. thermal calibration
> 
> 
> seems to work, step 3 was the magic sauce for me though! ;)

After the factory reset you don't need to run any gcode as it set identical values as are used in the E3D . hex files https://github.com/prusa3d/Prusa-Firmware/blob/v3.13.2/Firmware/thermal_model/e3d_REVO.h

The gcode files are for users that don't want to factory reset or got "bad" values at some point.

**Comment by slaesh** [comment]
I am pretty sure I did the factory reset first. And then doing the PID
calibration it started and immediately stopped and was in a weird state
afterwards. Seems like I did the factory reset with the old "normal"
firmware though 🤔🙈. Sh*t. Thanks for the hint. will start all over again
with a factory reset now and see what happens! ;) Sorry for the confusion 🤫

3d-gussner ***@***.***> schrieb am Do., 1. Feb. 2024, 13:57:

**Comment by jake-carpenter** [comment]
I just installed a REVO because it sounded like this has been sorted out, unfortunately I got errors immediately with v3.13.3-Revo.

* Printer: MK3S+. 
* Currently not in my enclosure, but I'll try it in the enclosure later.
* I'm doing my FW flashing from Octoprint, if that has any bearing.

Steps I have taken:

1. Flashed v3.11 and tried factory reset then PID calibration, as some old posts have recommended.
  a. Ran a print on v3.11 to confirm everything does work fine.
2. Flashed 3.13.3-Revo (7094).
3. Performed a full factory reset.
  a. I let the first-time setup run and calibrate everything. TM and PID were calibrated.
  b. At this point, prints would start but would not go for long before I got a thermal error.
4. Performed the Task 1 steps in #4105 (via Octoprint).
  a. Thermal error almost immediately after a print started.
5. Ran the `reset_revo.gcode` provided above from my SD card.
  a. Still have thermal errors within seconds of starting the print.

Not sure what else to do at this point. I'd be willing to re-check all the wiring for bad connections, except everything is fine on v3.11. In fact, it's printing on v3.11 as I write this. I'm happy to provide some logs if it will help here. I'll just need a link on what/how to pull them.

**Comment by 3d-gussner** [comment]
@jake-carpenter 

> I'm doing my FW flashing from Octoprint, if that has any bearing.

Please don't update your MK3/S/+ firmware with OctoPrint firmware flasher. It is still missing the extension for the MK3/S/+ language packs https://github.com/OctoPrint/OctoPrint-FirmwareUpdater/pull/325

Ensure that you flash the correct firmware version for the E3D Revo which is in the https://github.com/prusa3d/Prusa-Firmware/releases/download/v3.13.3/e3d_REVO_fw_MK3_MK3S_MK3S+_3_13_3_7094.zip
Looks like you did it in Step 2

The `reset_revo.gcode` writes the default E3D Revo TM and PID values to eeprom so you don't have to factory reset the printer. When you run this gcode you need to run PID and TM calibration again to get values for your printer.

Please also contact E3D Support for more help https://e3d-online.zendesk.com/hc/en-us/requests/new?ticket_form_id=11304553253917 check also their discord.

You can enable in OctoPrint save logfiles to get more information during the print. These logfiles should help E3D support to narrow down your issue.

**Comment by IndustrialCADWorks** [comment]
I just had this problem. Went back, went through everything listed above and other forums. What I might have figured out is on step 40 on the E3D instructions.

https://e3d-online.zendesk.com/hc/en-us/articles/4410028113169-Revo-Six-Prusa-MK3-MK3S-MK3S-upgrade-guide

It mentions PID tuning at 240 degrees if you plan to use materials other than PLA. When running the PID calibration through the wizard it runs at 230. I went back Calibrated PID to 240, ran through all the steps and no thermal issues or haults through the calibration. So I'm assuming you have to run the PID calibration prior to running the configuration wizard?? I don't know if that is true, but I know it worked for me.  Hope this helps.

**Comment by mhouse1** [comment]
> I just had this problem. Went back, went through everything listed above and other forums. What I might have figured out is on step 40 on the E3D instructions.
> 
> https://e3d-online.zendesk.com/hc/en-us/articles/4410028113169-Revo-Six-Prusa-MK3-MK3S-MK3S-upgrade-guide
> 
> It mentions PID tuning at 240 degrees if you plan to use materials other than PLA. When running the PID calibration through the wizard it runs at 230. I went back Calibrated PID to 240, ran through all the steps and no thermal issues or haults through the calibration. So I'm assuming you have to run the PID calibration prior to running the configuration wizard?? I don't know if that is true, but I know it worked for me. Hope this helps.

Does it still work after a week? In my experience, it'll work a few prints but eventually it starts erroring out during the middle of a print, you can restart and print again but with the risk of failure we reverted to 3.11

**Comment by IndustrialCADWorks** [comment]
I've been printing everyday and no problems, (pla, PE / CF, Nylon, PETG) multiple nozzle changes, no issues so far. Like I said not sure if it helps, but just posting my experience. If I have any issues I'll have to revert to 3.11, but for now no problems so far. Fingers Crossed...

**Comment by ndemarco** [comment]
@3d-gussner in reply to your [earlier comment](https://github.com/prusa3d/Prusa-Firmware/issues/4451#issuecomment-1881124015).

I am not printing in an enclosure, though I tried this also - same result.
I am running via Octoprint, but have tried printing from a card - same result.

E3D support had me go through some big steps to verify the integrity of the wiring and the electrical properties of their 'HeaterCore' part. I did this on video, posted [here](https://www.youtube.com/watch?v=Vqt04m8lj0o). In this video, the thermisistor responded exactly as specified in the [datasheet](https://www.mouser.com/datasheet/2/362/P18_NT_Thermistor-1535133.pdf). The heater's resistance matches E3D's specification, too.

The TM error occurs seconds after starting the run. I'm confused as to how the TM error is raised so quickly, when the thermisistor and heater are performing properly electrically.

One commentor on my video gave a detailed procedure to manually recalibrate. I plan to try that procedure on camera later, and will post a follow-up video.

**Comment by 3d-gussner** [comment]
@ndemarco This is a very detailed report and video, really appreciated. Makes troubleshooting sooo much easier.

Please try to connect your RPi (guess OctoPrint) enable serial logging in Octoprint and connect to the printer. Serial logs help a lot while troubleshooting.

You can use the https://github.com/prusa3d/Prusa-Firmware/files/12917795/reset_revo.gcode.txt (change filename extension to gcode) to reset to the default REVO PID and Thermal Model values.

To fine tune the PID and REVO on your machine
1. Let it cool down
2. LCD Calibration -> PID calibration
3. Let it cool down
4. LCD Calibration -> Thermal Model cal.

If that doesn't help please do a factory reset.

In your video you do a Factory reset BUT you only reset the language. You need to scroll down to "All Data" to reset everything.
Keep in mind that it will delete your statistics.

The comment from https://www.youtube.com/@n.neves1776 is good but only needed if you don't get it working with the steps above.

Why it may have failed on your printer:
I think that you
- had Thermal Model calibrated for the E3Dv6, which is stored in EEPROM
- have the PID values for the E3Dv6, which are stored in EEPROM
- then you changed to the E3D Revo and tried to run Thermal Model cal., which runs the TM calibration taking into account the previous stored values in EEPROM.
  - As the previous EEPROM values are for the E3Dv6 and these differ a lot from the E3D Revo the TM cal will complain that these outside expected thresholds and report the TM ERROR.

**Comment by 3d-gussner** [comment]
@ndemarco Your PID calibration in the Video probably TM ERROR kicked in and stopped.

The reset_revo.gcode contains the default values which are the same as in the factory reset and E3D Revo firmware version. This will reset any values you previously calibrated, so be careful with that.

**Comment by 3d-gussner** [comment]
@ndemarco I can understand the frustration, but the Thermal model saved already few users with broken cables and connectors.
The TM runs a simulation based on the calibrated values and compares it to the real data of the thermistor and power provided to the heater cartridge, so if the TM and PID values are from the E3Dv6 and the real data is from REVO it will report TM warnings and Errors.

**Comment by ndemarco** [comment]
@3d-gussner I appreciate all the feedback. I am carefully trying each of your points. In this video, I show the firmware flashing (which fails the first time around and succeeds the 2nd time). I then show the whooooooole setup, so you may want to >> for some. When it tries to TM calibrate, everything fails - STOPPED.

Now, I am waiting for the nozzle to cool completely. I will then run the reset_revo.gcode, followed by your [steps](https://github.com/prusa3d/Prusa-Firmware/issues/4451#issuecomment-2075305893) above - doing the PID calibration first.

**Comment by ndemarco** [comment]
The PID calibration completed normally. Unfortunately, the TM calibration fails within a few seconds of starting. For now, I reverted the firmware to 3.10.0, which is previous to the TM effort. I'm willing to help diagnose the issue with logs, etc. if it helps the community.

**Comment by mhouse1** [comment]
fwiw 3.11 is the  last stable version without the new thermal model, and really seems to be the only way forward if you want to use the much better revo six extruder.

**Comment by Sp00k1Gh0st** [comment]
Hi everyone I just wanted to share some information I found while diagnosing the thermal anomaly issue I was having with the revo. With mine I was finding that when the anomaly would happen I would have to disconnect the hot end from both jump wires and reconnect them before running the code provided above to reset everything successfully. I have found that a connection break is happening at the heater connection but not at the thermostat connection. After I cut the ends off of the jumper and hot end and soldered the wires together I was able to eliminate the problem all together and have been printing for over a week now without issue. I am sure there is more going on but hopefully this helps!

**Comment by 3d-gussner** [comment]
@Sp00k1Gh0st Thanks for sharing the information.
The thermal model triggered here correctly and you did the right thing: Look why it triggers and not just disable it. :clap: :medal_sports: 

As you can see the thermal model is not to annoy users but to improve the safety of the hotend, which can get very very hot. It detected the broken/loose/unreliable connection and @Sp00k1Gh0st investigated the issue to solve it.

This is another report I have seen that some connectors may cause issue. But it isn't only a REVO thing, we also have seen that some wires broke somewhere in the loom and only were "faulty" at specific positions or movements.

**Comment by ndemarco** [comment]
> @Sp00k1Gh0st Thanks for sharing the information. The thermal model triggered here correctly and you did the right thing: Look why it triggers and not just disable it. 👏 🏅
> 
> As you can see the thermal model is not to annoy users but to improve the safety of the hotend, which can get very very hot. It detected the broken/loose/unreliable connection and @Sp00k1Gh0st investigated the issue to solve it.
> 
> This is another report I have seen that some connectors may cause issue. But it isn't only a REVO thing, we also have seen that some wires broke somewhere in the loom and only were "faulty" at specific positions or movements.

This is most definitely not the issue with my Prusa + Revo printer. I've gone through the wiring and connections way more carefully than the Apollo 1 technicians did.

**Comment by acothebraco** [comment]
Like I say. I will no longer wait for prusa to solve my thermal problems with the MK3S+ in a enclosure. I sell my mk3s and switched to another printer. Important to know is that the mk3s is not designed to work in a enclosure.

**Comment by mhouse1** [comment]
I will give these [steps](https://github.com/prusa3d/Prusa-Firmware/issues/4451#issuecomment-2075305893) another try  once 3.14.0 hex is released; it was tagged 3 weeks ago but not built as hex yet. 

I will also try replacing the wires to the extruder if errors occur.

**Comment by ndemarco** [comment]
Latest update, [captured on video](https://youtu.be/-1ImEIG6ytw):

- Swapped Einsy Rambo board from a known good MK3
- Verified all wiring 2x
- Flashed firmware file E3DREVO_fw_MK3S_MK3S+_3_13_3_7094.hex
- Factory reset all (more than language)
- Begin wizard
- Observe all tests pass
- Observe 6+ beeps followed by 12 beeps.
- Observe small circle in lower left, with center dot.

Point me in a direction to solve this please.

**Comment by mhouse1** [comment]
the instructions https://github.com/prusa3d/Prusa-Firmware/issues/4451#issuecomment-2075305893 above isnt  clear but i assume you need to repeat these calibration steps after a factory reset; which you didnt do in your video, and im interpreting it as you dont need  to run the [reset_revo.gcode.txt](https://github.com/prusa3d/Prusa-Firmware/files/12917795/reset_revo.gcode.txt)  if you do the factory reset. 

```
To fine tune the PID and REVO on your machine

Let it cool down
LCD Calibration -> PID calibration
Let it cool down
LCD Calibration -> Thermal Model cal.
```

**Comment by ndemarco** [comment]
I don't even get to the point of running PID or TM. I get a series of beeps
and an o.

**Comment by gouryellas** [comment]
> Latest update, [captured on video](https://youtu.be/-1ImEIG6ytw):
> 
> * Swapped Einsy Rambo board from a known good MK3
> * Verified all wiring 2x
> * Flashed firmware file E3DREVO_fw_MK3S_MK3S+_3_13_3_7094.hex
> * Factory reset all (more than language)
> * Begin wizard
> * Observe all tests pass
> * Observe 6+ beeps followed by 12 beeps.
> * Observe small circle in lower left, with center dot.
> 
> Point me in a direction to solve this please.

downgrade to 13.11 until they figure things out. been using it for a long time with no problems on the revo

**Comment by Leopere** [comment]
Is this fixed in the latest version as of today?

**Comment by ndemarco** [comment]
̶I̶ ̶c̶a̶n̶ ̶d̶o̶ ̶s̶o̶m̶e̶ ̶t̶e̶s̶t̶i̶n̶g̶ ̶o̶v̶e̶r̶ ̶t̶h̶e̶ ̶w̶e̶e̶k̶e̶n̶d̶.̶
The above was a reply-to-email. I assumed there was a newer version released.

No, it is not fixed as of version 3.13.3.

Clarity rules over ambiguity; cite a version number in place of a relative reference like 'as of today'.

**Comment by Leopere** [comment]
I haven't had a thermal alert yet but I'm not sure what caused my first big one.

**Comment by ndemarco** [comment]
> I will give these [steps](https://github.com/prusa3d/Prusa-Firmware/issues/4451#issuecomment-2075305893) another try once 3.14.0 hex is released; it was tagged 3 weeks ago but not built as hex yet.
> 
> I will also try replacing the wires to the extruder if errors occur.

@mhouse1 Have you tried the RC 3.14.x yet? If you had good results, I'll also try it and report back.

**Comment by mhouse1** [comment]
> > I will give these [steps](https://github.com/prusa3d/Prusa-Firmware/issues/4451#issuecomment-2075305893) another try once 3.14.0 hex is released; it was tagged 3 weeks ago but not built as hex yet.
> > I will also try replacing the wires to the extruder if errors occur.
> 
> @mhouse1 Have you tried the RC 3.14.x yet? If you had good results, I'll also try it and report back.

3.14 is tagged but there are no hex files  included so I cant use it yet. https://github.com/prusa3d/Prusa-Firmware/releases/tag/v3.14.0

**Comment by Leopere** [comment]
Sounds close

**Comment by mhouse1** [comment]
new firmware is ready, i will test this on July 4th and report back, looking forward to getting this resolved and especially the new reprint menu feature; it will really help with running the farm of 100's of files.

**Comment by mhouse1** [comment]
> new firmware is ready, i will test this on July 4th and report back, looking forward to getting this resolved and especially the new reprint menu feature; it will really help with running the farm of 100's of files.

So far its working great on the 3 printers I updated a week ago; fw3.14 for mk3s and mk3s+ with revo.

 I think the main difference was running the reset_revo.gcode , I didnt have to factory reset any printer. I tried this on printers as old as 2019 with 982days of print time and new printer built in late 2023 with 40days of print time. 

The older printer starts thermal model cal right away after firmware update, newer one didnt. printer number 49 was stuck at PID cal 0/5 but I just restarted the printer and tried again.

Details:
```
====================================
printer #50 built 9/10/23
flash new mk3s revo hex firmware 3.14
shows prusa i3 MK3S+R ok on LCD

printer  statistics total filament 2766.26m
total print time 40d


use reset_revo.gcode on sd card
response: file incomplete continue anyway?
yes
Press to reset REVO
Press to Save
0 hours 1 minutes

next, performed PID  calibration to 250

thermal model calibration
TM: initial C  est.

====================================
printer #8 built 9/21/19
flash new mk3s revo hex firmware 3.14
after flashing it automatically started thermal model cal
shows prusa i3 MK3S+R ok on LCD
52453.14m filament, 
total print time 982d

use reset_revo.gcode on sd card
response: file incomplete continue anyway?
yes
Press to reset REVO
Press to Save
0 hours 1 minutes

next, performed PID  calibration to 250

thermal model calibration

====================================
printer #49 built 9/10/23
print meters, days (5781.49m,86 days)
ran reset_revo.gcode
continue anyway? > yes
Press to reset REVO 
press to save

flash new mk3s revo hex firmware 3.14

next, performed PID  calibration to 250
pid cal 0/5 stuck at 31C extruder 26c bed
reset the printer
ran reset_revo.gcode
retry PID cal to 250C

next, thermal model cal

```

We'll start updating the rest of the printers and report back if issues.

**Comment by Leopere** [comment]
awesome I haven't been having trouble with mine either on this version but I was waiting because when I had a previous version I didn't have trouble until I did and I don't really know what had changed so I wanted to hear it from someone else before saying anything.

**Comment by 3d-gussner** [comment]
@mhouse1 Thanks for sharing the information :hugs: . Always nice to see that kind of feedback. :crossed_fingers: The rest of the printers updates are also going that "smoothly"

**Comment by capsulecorplab** [comment]
Been running 3.14.0 firmware for the Mk3S with Revo for a couple of days now. Seems to be working so far without errors!

Though, the option to load filament when autoload is turned off seems to have disappeared. Should I open an issue for this?

**Comment by 3d-gussner** [comment]
@capsulecorplab Thanks for the positive feedback.

The "Load filament" and "Unload filament" menus are shown depending on the state of the filament sensor.
- When filament is detected and printer is idle "Unload filament" is shown and "Load filament" is hidden.
- When NO filament is detected and printer is idle "Load filament" is shown and "Unload filament" is hidden.
- When the FSensor is disabled both "Load filament" and "Unload filament" are shown as the printer doesn't "know" if there is any filament and the user has to be able to use both menus.

Hope that it makes clear.

If you have an question or issue with it, then please open a new Issue to keep this one "clear" to one topic.

**Comment by mhouse1** [comment]
So far I  have 4 printers running new fw3.14 successfully 
Two printers are getting thermal anomaly with firmware 3.14 after updating and is unable to perform PID cal. One printer I also tried factory reset to no help. Today, the print farm is running so the room is hot, maybe this is interfering with calibration. I will try this again later when print farm is off. 

```
====================================
printer #10 built 12/16/19
print meters, days (44092.22m,945 days)
ran reset_revo.gcode
file incomplete continue anyway? > yes
Press to reset REVO 
press to save

flashed fw 3.14 thermal model cal
 starts automatically after flashing
fails right away, menu shows Clear TM error above print from sd
cleared TM error

start PID cal to 250C
stuck at nozzle 50/0 bed 28/0 PID cal. 0/5
nozzle is cold
performed power cycle
when i flipped powering switch extruder moves left for some reason
when powering on it tries to resume a print
restart printer using LCD X button
detects thermal model not set
click lcd to continue tries resuming print again, use menu to "Stop Print"
restart printer using LCD X button again
detects thermal model not set, but  does not try to resume print
use reset_revo.gcode
start PID cal to 250C
stuck at nozzle 52/0 bed 28/0 PID cal. 0/5
Perform factory reset all data using instructions at (https://help.prusa3d.com/article/full-system-refresh-original-prusa-i3_133258)
start PID cal to 250C
stuck at nozzle 58/0 bed 28/0 PID cal. 0/5
removed sd card
start calibration > wizard
while extruder is up, realize revo nozzle is a little loose, tightened it
wizard  starts searching bed calibration point
wizard completes and starts thermal model cal
thermal model cal results in thermal anomaly within 1 second
try PID cal to 250 again
PID cal ramps nozzle to 80C then stuck at nozzle ~52/0 bed 28/0 PID cal. 0/5
====================================
printer #40 built 4/26/21
print meters, days (m, days)
ran reset_revo.gcode
file incomplete continue anyway? > yes
Press to reset REVO 
press to save

flashed fw 3.14 thermal model cal
 starts automatically after flashing
next, performed PID  calibration to 250
cannot complete PID cal stuck at PID cal. 0/5
when trying to print from SD LCD Shows PAUSED THERMAL ERROR
restored to fw 3.11 for now so i can use the printer
performed PID cal to 250 and started a print
```

**Comment by mhouse1** [comment]
so far 2 out  of 9 printers can't be updated to fw3.14 due to PID cal and thermal model failing right away even though both printers are functional on fw3.11 and is used everyday . Some inconsistencies, some printers starts thermal model cal right away after flashing fw3.14, some doesn't, printer number 10 built 12/16/19 I reverted back to fw3.11  because it kept failing to calibrate on fw3.14 even after factory reset and reset_revo.gcode, when I tried flashing fw3.14 again a few days later it starts self test after flashing. 

for now any printer that accepts fw3.14 I'm updating to that otherwise they get updated to fw3.11 

Details:
```

====================================
printer #10 built 12/16/19 attempting fw upgrade again
has fw 3.11
after flashing 3.14  it starts self test ("First, I will run the selftest to check most common assembly….)
starts thermal cal
fails right away with LCD showing print aborted
run reset revo.gcode
start PID cal, cannot complete PID cal  LCD shows PID cal. 0/5 LCD  menu cannot scroll
reset using LCD button
revert to fw 3.11

====================================
printer #35 built 1/15/21
has fw 3.11
print meters, days (?m, ?days)
after flashing 3.14 it starts thermal cal right away
successful
next, performed PID  calibration to 250
successful

====================================
printer #39 built 4/26/21
has fw 3.11
print meters, days (23681.91m, 386days)

after flashing 3.14 it starts thermal cal right away
next, performed PID  calibration to 250
started small print 33min
successful
====================================
printer #42 built 5/3/21
has fw 3.11
after flashing 3.14 does not auto thermal cal , LCD shows PRUSA i3 MK3S+R Ok.
Start PID cal to 250 … completes successfully
wait for cool down
next, performed thermal model cal
successful
```

**Comment by capsulecorplab** [comment]
> Been running 3.14.0 firmware for the Mk3S with Revo for a couple of days now. Seems to be working so far without errors!
> 
> Though, the option to load filament when autoload is turned off seems to have disappeared. Should I open an issue for this?

I spoke too soon. I forgot I had disabled thermal model protection while running v3.13.2 before updating to v3.14.0. Factory resetting the printer brought back the thermal anomaly error until I reran the gcode to disable thermal model protection.

**Comment by 3d-gussner** [comment]
@capsulecorplab Thanks for the feedback. Did the TM cal after the factory reset work?

**Comment by capsulecorplab** [comment]
> @capsulecorplab Thanks for the feedback. Did the TM cal after the factory reset work?

I tried running TM cal after the reset, but couldn't get the thermal anomaly error to go away, so I ended up just disabling the TM protection

**Comment by mhouse1** [comment]
> > @capsulecorplab Thanks for the feedback. Did the TM cal after the factory reset work?
> 
> I tried running TM cal after the reset, but couldn't get the thermal anomaly error to go away, so I ended up just disabling the TM protection

which one is safer, disabling TM protection and using fw3.14 or using fw3.11 ?

**Comment by capsulecorplab** [comment]
> which one is safer, disabling TM protection and using fw3.14 or using fw3.11 ?

I don't recall TM issues with fw3.11, but I also haven't printed on that fw enough to tell. I think I've had the best experience with fw3.13.2 (with TM protection disabled) so far - in part due to the fact that fw3.11 lacks filament runout sensing and fw3.14's "load filament" option seems to disappear for me when autoload is disabled, granted that's a separate issue.

**Comment by mhouse1** [comment]
no TM issues with fw3.11 but I thought the TM that was added after 3.11 improves on detecting TM anomalies, I wonder if disabling the new TM means we still have whatever safety feature that was in 3.11. Theres definitely lots of improvements in 3.14 that I'd like to use  but if disabling the TM model means its less safe than 3.11 then I won't risk it.

With TM model disabled in 3.14 can it still detect extruder thermal runaway events?  3.11 has been great in detecting thermal runaways which I've see happen 8 times so far in the last 4 years on a print farm of 50+ mk3s printers.

**Comment by 3d-gussner** [comment]
@mhouse1 Sorry for the late answer.

>With TM model disabled in 3.14 can it still detect extruder thermal runaway events?

Yes the default thermal protection is always active. 
With the TM active, but you might not see it as the TM reacts in seconds while thermal runaway takes longer.

Did you contact E3D support as they should be able to help you with REVO and TM issues,

**Comment by ndemarco** [comment]
As you might imagine, [E3D ran out of ideas, and directed me to contact Prusa for support](https://en.wikipedia.org/wiki/Buck_passing).

E3D as a supplier of retrofit parts is ultimately responsible for making their parts are compatible.

I have offered to assist by providing logged data if properly instructed.

![image](https://github.com/user-attachments/assets/ed4b5b78-60b3-41a4-8575-9111675054f8)

**Comment by mhouse1** [comment]
> Yes the default thermal protection is always active.

I think the confusion from the community is fw3.11 was such a stable fw (no thermal issues in our years of use) why mess with it when it introduces complexity such as having to manage multiple fw builds. And the fact that a printer that prints completely fine on fw3.11 might experience odd thermal anomalies when using fw3.14: in some cases, a "PAUSED thermal anomaly" will appear on the LCD and prevent the print from starting but a reboot fixes it, in other cases "thermal anomaly" appears during print start with audible alarm but does not prevent print. Although a majority of the printers running fw3.14 are running good, it will take me a few more months of runtime to conclude whether updating from 3.11 to 3.14 was worth it.


I will try disabling thermal protection on fw3.14 just for those printers (2 out of 12) that are completely functional on fw3.11 but errors out ASAP on fw3.14 preventing the printer from being used.

**Comment by 3d-gussner** [comment]
> > Yes the default thermal protection is always active.
> 
> I think the confusion from the community is fw3.11 was such a stable fw (no thermal issues in our years of use) why mess with it when it introduces complexity such as having to manage multiple fw builds. And the fact that a printer that prints completely fine on fw3.11 might experience odd thermal anomalies when using fw3.14: in some cases, a "PAUSED thermal anomaly" will appear on the LCD and prevent the print from starting but a reboot fixes it, in other cases "thermal anomaly" appears during print start with audible alarm but does not prevent print. Although a majority of the printers running fw3.14 are running good, it will take me a few more months of runtime to conclude whether updating from 3.11 to 3.14 was worth it.
> 
> I will try disabling thermal protection on fw3.14 just for those printers (2 out of 12) that are completely functional on fw3.11 but errors out ASAP on fw3.14 preventing the printer from being used.

@mhouse1 Thanks for the feedback and sorry to hear that 2 of your 12 printers keep be difficult with TM enabled.

The thermal model already proved that it is useful as some users had broken or semi broken thermistor or heater cables and found the issue. It reacts to issues way faster that the regular thermal protection but is also way more "sensitive" which can cause false positive warnings and errors.

For the stock E3D V6 we had tons of data and test machines internally and I think that the values are working great.

For the REVO we had just few community members submitting their values and E3D provided some other values for the REVO within their performance tolerances.

@mhouse1 Did you try to raise the `W` and `E` values?

**Comment by 3d-gussner** [comment]
> As you might imagine, [E3D ran out of ideas, and directed me to contact Prusa for support](https://en.wikipedia.org/wiki/Buck_passing).
> 
> E3D as a supplier of retrofit parts is ultimately responsible for making their parts are compatible.
> 
> I have offered to assist by providing logged data if properly instructed.
> 
> ![image](https://private-user-images.githubusercontent.com/6326614/356236042-ed4b5b78-60b3-41a4-8575-9111675054f8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQxMzExNDIsIm5iZiI6MTcyNDEzMDg0MiwicGF0aCI6Ii82MzI2NjE0LzM1NjIzNjA0Mi1lZDRiNWI3OC02MGIzLTQxYTQtODU3NS05MTExNjc1MDU0ZjgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgyMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MjBUMDUxNDAyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZmEyYTM5OTNiNWQyMTM5MTQ1NjMwYzEyZTU2OGI0NjhhYTFhNjQ5MGY1Njg1OTJmNTIyMjlmNGVhZGFlZmNhZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ._TTVf9XncC6pxvl1WIADyDro-YALYc3g_CokMnoi-Zs)

Hi, sorry to hear that E3D couldn't help you.

My time on this 3rd party issue is very limited but please try https://github.com/prusa3d/Prusa-Firmware/issues/4105 and provide there your results?

**Comment by mhouse1** [comment]
> The thermal model already proved that it is useful as some users had broken or semi broken thermistor or heater cables and found the issue. It reacts to issues way faster that the regular thermal protection but is also way more "sensitive" which can cause false positive warnings and errors.

thank you for all the work put into this , I do believe I have some bias towards changing from fw3.11

We have some printers on fw3.14 that throws thermal anomally errors during startup, perhaps the firmware response is a sign that the printer wires or revo is wearing out, or its caused by a draft at that corner of the room and moving printer location will improve it, time will tell.

**Comment by 3d-gussner** [comment]
Draft can cause some TM issues. I expected the REVO having less issues with draft because of the design of it. Or are these printers next to an walk-in freezer with -40°C? :wink:

If I remember correctly testing the TM on REVO wasn't as easy to trigger as with the E3D v6 see as an example https://github.com/prusa3d/Prusa-Firmware/issues/4105#issuecomment-1483476002

**Comment by mhouse1** [comment]
I ended up updating all printers to fw3.14 and disabled thermal model on more than half of the printers (12+ printers); due to thermal model either being annoying or preventing prints. No issues so far operating this way. For most printers with thermal model disabled they on average had 30000 meters of filament and 500 to 600 total print time days (from printer's LCD statistics page).

**Comment by mhouse1** [comment]
update: we started reverting to fw3.11

on the initial two printers that I disabled thermal model on, after a few months of using fw3.14 prints wont start, now it errors out with thermal error right a way. Reverting to 3.11 allowed printing again. 

Observed something dangerous on a separate fw3.14 revo printer (2021 mk3s) this morning; this printer also did not have thermal model disabled. a 12hr print was stuck overnight 80% finished with the printer not moving,  mid print, with fully heated nozzle, filament, and bed; im not sure how long it was stuck like this but seems rather dangerous, the LCD displayed no signs of error messages. I think safety should be added that if motors are not being moved  for some time disable heating.

some other random fw3.14 issues:
* sometimes the nozzle size setting changes on its own, a bit annoying but doesnt cause issues
* theres more sd and fw crash issues, maybe our sd card is  wearing out but i did not see as many  on fw3.11

**Comment by 3d-gussner** [comment]
@mhouse1 
>this printer also did not have thermal model disabled. a 12hr print was stuck overnight 80% finished with the printer not moving, mid print, with fully heated nozzle, filament, and bed; im not sure how long it was stuck like this but seems rather dangerous, the LCD displayed no signs of error messages. I think safety should be added that if motors are not being moved for some time disable heating.

Was it a SD card print or a host print like from Octoprint?

**Comment by mhouse1** [comment]
> @mhouse1
> 
> > this printer also did not have thermal model disabled. a 12hr print was stuck overnight 80% finished with the printer not moving, mid print, with fully heated nozzle, filament, and bed; im not sure how long it was stuck like this but seems rather dangerous, the LCD displayed no signs of error messages. I think safety should be added that if motors are not being moved for some time disable heating.
> 
> Was it a SD card print or a host print like from Octoprint?

this was printing from SD card

**Comment by JeffWDH** [comment]
For better or for worse, the only way I could get my Revo 6 (40 watt) to work in anything later than 3.11 on my MK3S was to disable the thermal model using a serial console. I did a factory reset and it would always fail when the nozzle was heated up. 
```
M310 S0
ok
M500
echo:Settings Stored
ok
```
My printer works flawlessly on 3.11. I need to do some testing on whether it actually works well on 3.14.1 or if it just passed the initial wizard. 

For completeness, this is the error I get from the console:
```
TM: calibr. failed!
echo:Thermal Model settings:
echo:  M310 I0 R25.30
echo:  M310 I1 R23.90
echo:  M310 I2 R22.50
echo:  M310 I3 R19.60
echo:  M310 I4 R19.00
echo:  M310 I5 R18.30
echo:  M310 I6 R17.70
echo:  M310 I7 R17.10
echo:  M310 I8 R16.80
echo:  M310 I9 R16.50
echo:  M310 I10 R16.30
echo:  M310 I11 R16.00
echo:  M310 I12 R15.90
echo:  M310 I13 R15.70
echo:  M310 I14 R15.60
echo:  M310 I15 R15.40
echo:  M310 P40.00 U-0.0014 V1.05 C8.77 D0.1500 L270 S1 B0 E1.23 W0.85 T-7.00
TM: error |1.315643|>0.850000
TM: error triggered!
//action:cancel
echo:Printer stopped due to errors. Supervision required.
```

So far I've done a Calicat in PLA and it turned out as I would expect with no errors. I'd love to re-enable the new thermal model functionality but for now this is working.

**Comment by mhouse1** [comment]
> For better or for worse, the only way I could get my Revo 6 (40 watt) to work in anything later than 3.11 on my MK3S was to disable the thermal model using a serial console. I did a factory reset and it would always fail when the nozzle was heated up.
> 

thats what I observed as well but after a few months of continuous printing it stopped working and would immediately error out when print starts, but reverting to 3.11 it functions and prints parts  just fine. If it was truly an thermal issue i'd expect the parts to have defects

**Comment by JeffWDH** [comment]
I suspect hardware failure in my case. I was printing tonight and had a thermal runaway event and could smell a hot electric smell coming from the hotend. When looking for a new one I noticed the Prusa 40w heater cores all now come with blue thermistor wires and mine is white. I suspect mine was one of the early units, potentially other people with this issue may have similar hardware? 

After further investigation, while doing a PID calibration the same thing happened and the temperature reading stalled around 180. After jiggling the thermistor wire near the thermistor the temperature reading jumped to 300(!) at which point I aborted. 

I contacted E3D support with my findings and they confirmed my heater core was an old model and offered to send out a replacement. I will update this comment when I install it and retry the thermal model code.

**Comment by 3d-gussner** [comment]
@JeffWDH Thanks for sharing, glad nothing happened to you and your printer. The thermal model triggered a warning and you did the right thing to investigate why and not just disable the TM.

**Comment by JeffWDH** [comment]
Update: After installing a new (blue thermistor cable) heater core I was able to successfully perform a PID tune and calibrate the thermal model. I still have to do some test prints when I have time to babysit the printer but so far it's an improvement.
```
start
echo: 3.14.1-8237_74a577bc0
MMU is Off
SpoolJoin is Off
echo: Last Updated: 2024-09-24 14:54:47 | Author: prusa3d
echo: Free Memory: 2509  PlannerBufferBytes: 1760
echo:Stored settings retrieved
adc_init
Hotend fan type: NOCTUA
CrashDetect ENABLED!
Sending 0xFF
echo:SD card ok
//action:dump_available
echo:enqueing "M303 E0 S210"
LCD status changed
PID Autotune start
T:23.75 @:127
T:33.28 @:127
T:45.50 @:127
T:55.55 @:127
T:65.00 @:127
T:75.15 @:127
T:83.39 @:127
T:92.31 @:127
T:99.70 @:127
T:107.09 @:127
T:115.35 @:127
T:122.26 @:127
T:129.96 @:127
T:136.68 @:127
T:144.18 @:127
T:150.37 @:127
T:156.69 @:127
T:163.66 @:127
T:169.15 @:127
T:176.37 @:127
T:181.96 @:127
T:187.73 @:127
T:194.28 @:127
T:199.28 @:127
T:205.54 @:127
T:211.30 @:0
T:215.47 @:0
T:214.27 @:0
T:211.61 @:0
T:208.57 @:127
T:209.55 @:127
T:215.73 @:0
T:219.69 @:0
T:219.69 @:0
T:216.82 @:0
T:214.22 @:0
T:210.83 @:0
 bias: 84 d: 84 min: 208.04 max: 219.79
T:208.62 @:84
T:209.38 @:84
T:213.02 @:0
T:214.53 @:0
T:213.18 @:0
T:210.78 @:0
 bias: 75 d: 75 min: 208.39 max: 214.69
T:208.21 @:75
T:209.73 @:75
T:211.88 @:0
T:213.07 @:0
T:210.99 @:0
 bias: 72 d: 72 min: 208.21 max: 213.18
 Ku: 36.94 Tu: 10.80
 Classic PID
 Kp: 22.17
 Ki: 4.10
 Kd: 29.92
T:208.13 @:72
T:208.35 @:72
T:210.00 @:72
T:212.66 @:0
T:210.78 @:0
 bias: 72 d: 72 min: 207.86 max: 212.66
 Ku: 38.20 Tu: 10.26
 Classic PID
 Kp: 22.92
 Ki: 4.47
 Kd: 29.40
T:208.44 @:72
T:208.39 @:72
T:210.00 @:72
T:212.14 @:0
T:211.04 @:0
 bias: 72 d: 72 min: 207.86 max: 212.14
 Ku: 42.86 Tu: 10.26
 Classic PID
 Kp: 25.71
 Ki: 5.01
 Kd: 32.98
PID Autotune finished! Put the last Kp, Ki and Kd constants from above into Configuration.h
LCD status changed
echo:enqueing "M301 P25.71 I5.01 D32.98"
echo:enqueing "M500"
ok p:25.71 i:5.01 d:32.98
echo:Settings Stored
LCD status changed
M310 S1
ok
M500
Command not found!
echo:Settings Stored
ok
echo:enqueing "G28 W"
echo:enqueing "G1 X125 Y105 Z1 F8000"
tmc2130_home_enter(axes_mask=0x04)
tmc2130_home_exit tmc2130_sg_homing_axes_mask=0x04
tmc2130_home_enter(axes_mask=0x01)
echo:busy: processing
  0 step=62 mscnt= 992
tmc2130_goto_step 0 45 2 1000
tmc2130_home_exit tmc2130_sg_homing_axes_mask=0x01
tmc2130_home_enter(axes_mask=0x02)
  0 step= 1 mscnt=  31
tmc2130_goto_step 1 33 2 1000
tmc2130_home_exit tmc2130_sg_homing_axes_mask=0x02
echo:busy: processing
echo:busy: processing
echo:enqueing "M310 A F1"
TM: calibration start
LCD status changed
TM: cool down <50C
TM: initial C est.
TM iter:0 v:10.73 e:0.330
TM iter:1 v:8.54 e:0.256
TM iter:2 v:8.54 e:0.158
TM iter:3 v:8.54 e:0.098
TM iter:4 v:8.54 e:0.060
TM iter:5 v:8.54 e:0.037
TM iter:6 v:8.54 e:0.023
TM iter:7 v:8.54 e:0.014
TM iter:8 v:8.54 e:0.009
TM: initial R 230C
TM iter:0 v:22.19 e:0.479
TM iter:1 v:22.19 e:0.296
TM iter:2 v:22.19 e:0.183
TM iter:3 v:22.19 e:0.113
TM iter:4 v:22.19 e:0.070
TM iter:5 v:22.19 e:0.043
TM iter:6 v:22.19 e:0.027
TM iter:7 v:22.19 e:0.016
TM iter:8 v:22.41 e:0.010
TM iter:9 v:22.41 e:0.006
TM: cool down <50C
TM: refine C est.
TM iter:0 v:10.73 e:0.330
TM iter:1 v:8.54 e:0.256
TM iter:2 v:8.54 e:0.158
TM iter:3 v:8.54 e:0.098
TM iter:4 v:8.54 e:0.060
TM iter:5 v:8.54 e:0.037
TM iter:6 v:8.54 e:0.023
TM iter:7 v:8.54 e:0.014
TM iter:8 v:8.47 e:0.009
TM: R[15] estimate.
TM iter:0 v:15.76 e:0.261
TM iter:1 v:15.76 e:0.161
TM iter:2 v:14.19 e:0.111
TM iter:3 v:13.22 e:0.073
TM iter:4 v:13.22 e:0.045
TM iter:5 v:13.59 e:0.027
TM iter:6 v:13.59 e:0.017
TM iter:7 v:13.45 e:0.011
TM iter:8 v:13.45 e:0.007
TM: R[11] estimate.
TM iter:0 v:15.76 e:0.261
TM iter:1 v:15.76 e:0.161
TM iter:2 v:14.19 e:0.111
TM iter:3 v:14.19 e:0.068
TM iter:4 v:14.19 e:0.042
TM iter:5 v:13.82 e:0.027
TM iter:6 v:13.82 e:0.017
TM iter:7 v:13.96 e:0.010
TM iter:8 v:13.96 e:0.006
TM: R[7] estimate.
TM iter:0 v:15.76 e:0.261
TM iter:1 v:15.76 e:0.161
TM iter:2 v:14.19 e:0.111
TM iter:3 v:14.19 e:0.068
TM iter:4 v:14.79 e:0.041
TM iter:5 v:14.79 e:0.025
TM iter:6 v:14.79 e:0.015
TM iter:7 v:14.93 e:0.009
TM: R[3] estimate.
TM iter:0 v:15.76 e:0.261
TM iter:1 v:15.76 e:0.161
TM iter:2 v:15.76 e:0.100
TM iter:3 v:16.73 e:0.058
TM iter:4 v:16.73 e:0.036
TM iter:5 v:16.73 e:0.022
TM iter:6 v:16.96 e:0.014
TM iter:7 v:16.96 e:0.008
LCD status changed
echo:Thermal Model settings:
echo:  M310 I0 R22.41
echo:  M310 I1 R21.05
echo:  M310 I2 R19.69
echo:  M310 I3 R16.96
echo:  M310 I4 R16.46
echo:  M310 I5 R15.95
echo:  M310 I6 R15.44
echo:  M310 I7 R14.93
echo:  M310 I8 R14.69
echo:  M310 I9 R14.45
echo:  M310 I10 R14.21
echo:  M310 I11 R13.96
echo:  M310 I12 R13.84
echo:  M310 I13 R13.71
echo:  M310 I14 R13.58
echo:  M310 I15 R13.45
echo:  M310 P40.00 U-0.0014 V1.05 C8.47 D0.1500 L270 S1 B0 E1.23 W0.85 T-7.00
echo:enqueing "M500"
echo:Settings Stored
```

**Comment by DracoSeeker** [comment]
This seems to have worked for my prusa mk3s+. It's been about 3 days and no issues with 3.14.1.

**Comment by mhouse1** [comment]
we're  finding that ~20% of our MK3S printers with revo six cannot use 3.14.1 but have been functioning just fine at 3.11 , we've been operating this way for a few months now.

**Comment by ndemarco** [comment]
> we're finding that ~20% of our MK3S printers with revo six cannot use 3.14.1 but have been functioning just fine at 3.11 , we've been operating this way for a few months now.

What is the total population you're referring to: MK3S with Revo 6?

Have you observed what differentiates those printers that work with >=3.14 and those that only work with =3.11?

**Comment by mhouse1** [comment]
about 20 mk3s printers still operational. I suspect that the the thermal model doesn't account for all revo six units, especially the earlier units, i'll know with more time, we have some new revo six units that arrived December 2024.

It's amazing  that the 4 to 5 year old mk3s machines are even operational, but we're starting to see the cables wear out and motors stutter (I originally thought it was bearings but using just the naked belt  to spin shaft while motor is unplugged I can feel a little stutter). The MK3S's served  us well, the amount of time required to debug and fix them by pulling parts from other printers is not cost effective. We've started to  retire the MK3S machines  and replacing them with MK4S machines.

**Comment by jkon** [comment]
I just tried 3.14.1 and the Thermal Anomaly error came up about 5 seconds after trying to preheat. Interestingly, it still did the first layer calibration when prompted after flashing it and went through as if the hot end was heated even though it stopped heating after the error. Maybe my MK3S+ is also too old. I bought it second hand from someone who was replacing it with something bigger and faster for his etsy business. Maybe I just need to make the jump to the MK4S too. I just wish I had gotten more use out of it with the REVO. I love how easy it is to change the nozzles

**Comment by agryson** [comment]
I am also having this issue, but can calibrate and even print for several hours but in a given 24 hour period will have 2 to 3 such thermal errors. Frustratingly it says "paused due to thermal error" but even though I am sitting beside it all day and can verify there are no issues, or connection problems, I can not resume...
Note: there have been no hardware changes, this issue has only appeared since upgrading the firmware.

**Comment by ndemarco** [comment]
I'm persistent to a fault, yet I've given up on the E3D Revo. The MK3S is
obsolete. I also tried the Revo on a Mk3.5 converted printer. That was
another waste of time. I'm done with the Revo.

**Comment by agryson** [comment]
I tried again followed by the below procedure (which is, I admit, the same as a careful reading of the OP) and just completed a nearly 3 hour print with no apparent issues. I have just launched an 8 hour one and will report back.

## Prerequisites
> Use the Revo version of the firmware!
    [Firmware Page](https://help.prusa3d.com/downloads/mk3s)
    Scroll down and find the Revo version

Proper file will be named using this convention:
MK3S_MK3S+-E3DREVO_FW_x.xx.x_MULTILANG.hex

## Post-Install of firmware
1. Reset Prusa for Revo
    LCD-menu > Print from SD Card > Maintenance > reset_revo.gcode <- depnds on where you put it on your SD Card
2. Run a PID calibration
    LCD-menu > Calibration > PID calibration
3. Run a Thermal Model Calibration
    LCD-menu > Calibration > Thermal Model cal.
4. Run a First Layer Calibration
    LCD-menu > Calibration > First Layer Calibration

## Notes
Previous attempts I either had the wrong firmware (not the Revo version)
OR
had reset after the calibration rather than before.

Making sure I did them in this order (Proper firmware _then_ reset _then_ calibrate) has at least gotten one build off the plate.

I'll be back tomorrow to report if this build finished.

**Comment by 3d-gussner** [comment]
@agryson @ndemarco Please contact E3D support. If the REVO thermal model values aren't correct they should request/provide new values.
Remember that we only compile the E3D-Revo community firmware as E3D is a long term partner of Prusa. E3D is responsible for the support and correct values.

There were only few REVO users who helped to implement the REVO with thermal model see here https://github.com/prusa3d/Prusa-Firmware/issues/4105

@agryson Glad to hear that you got it working. The E3D REVO is a mod and need more attention to get it running if you come from the Original Prusa MK3S (previous firmware).
BTW When you "factory reset" the printer, with E3D revo firmware flashed, the default E3D Revo thermal model values.

**Comment by agryson** [comment]
I can happily report that for me my 8 hour print seems to have had 0 thermal warnings. 🥳

@3d-gussner I understand and agree that any issues with the thermal models (or even installation of the custom firmware) are more for E3D.

However, at least in my case, while getting _out_ of the problem is perhaps an E3D issue, getting _into_ the problem could be improved by Prusa - in my case PrusaSlicer notified me of a firmware update and pointed me here:
![Image](https://github.com/user-attachments/assets/e9435930-b2e4-45ec-8720-4dcbd93e5d5c)

Also, I had seen in earlier  release notes (v13.3.0) that Prusa 'supported' Revo so I had assumed a custom firmware was no longer needed and didn't even check github.

If Revo specific firmware is still needed, even after v3.13.0 (which it seems to be in my case) I think at least clearer messaging around this from Prusa would be helpful and reduce support strain.

**Comment by agryson** [comment]
Ah, I seem to have spoken too soon, a thermal anomaly error has just stopped a print again. I will be reaching out to E3D.

**Comment by 3d-gussner** [comment]
@agryson I don't think that there will be any update on the official Prusa website. As a REVO user you should look at the E3D website for any firmware updates.

If you have the option to monitor the printer via a serial USB connection that would help a lot to see what happens.
For example some models have a shape that the part cooling fan airs "blows back" on the nozzle and causes unexpected temp drop.

Adding `M155 S1 C3` to the start gcode, will report every second the temperature and fan speed.
`M155 S1 C7` will add another report of the position of the printer. This might be helpful if you have a broken cable or connection.

I know a user who had the REVO connector rubbing of the insulation of the cable and the Thermal Error only occurred at very specific location / movement. Please check your wires especially close to the REVO connectors.

Preheat the nozzle and slowly move the cable loom to see if you have any bad connection. Move the extruder from left to right and back, lift the Z and try moving again.

Is there any specific pattern you can see when the Thermal model errors out? BTW you can Clear the TM error and try to continue to print.

More information about the Thermal model gocde [M310](https://reprap.org/wiki/G-code#M310:_Temperature_model_settings)

Hope that helps.

**Comment by agryson** [comment]
Thank you, yes I understand and have updated my own upgrade documentation to account for it. 
I have contacted E3D and will try to sort the issue with them.

**Comment by getgray** [comment]
I got a REVO Six Prusa Edition kit for my MK3S+ from Printed Solid about a year ago, and just installed it new out of the box this week.  FWIW the printer is like new, it has printed less than 7 spools of filament.  I am running the stock REVO brass 0.4 nozzle.  When printing I'm getting the “Thermal Anomaly” error.  Coincident with the installation, I loaded firmware MK3S_MK3S+-E3DREVO_FW_3.14.1_MULTILANG.hex.  I did a factory reset after loading the firmware and installing the REVO and then went through the whole factory calibration wizard.  My printer is not in an enclosure.  It is not in a drafty or cold room.  Being aware of this issue I  tried to get ahead of it by following the steps described.  I understand there were some REVO units with a faulty thermistor, I don't know how to tell if I got one.  My printer gets the error shortly after starting the first layer and beeps, but so far it has continued to print, the error remains on the LCD screen throughout.  I am aware of, and have the “reset_revo” GCODE which I have not run yet, but I thought the factory reset would have taken care of that?  I've opened a support case with E3D but is there anything I missed or should do to correct meanwhile?

**Comment by agryson** [comment]
@getgray I believe the RESET_REVO is a required step, regardless of factory reset.

If it can help, below you'll find my internal documentation (an amalgam of the official docs and what I've found here and other places).
My issue came from poor connections, as @3d-gussner said in their latest comment, I resolved that with E3D support.

I have had no further issues after following this procedure (especially the Post-Install steps) with verified cabling.
Obviously adapt to wherever you store your maintenance gcodes.

## Firmware Updates
### Prerequisites
> [!WARNING]
> Use the Revo version of the firmware! [[Firmware Page](https://help.prusa3d.com/downloads/mk3s)](https://help.prusa3d.com/downloads/mk3s)
 
Proper file will be named using this convention:`MK3S_MK3S+-E3DREVO_FW_x.xx.x_MULTILANG.hex`

### Procedure
1. Connect the RAMBo board using the square-shaped USB-B 2.0 cable. The printer must be ON!
2. Unzip the firmware from the .zip-file
3. Open *PrusaSlicer*, navigate to Menu -> Configuration -> Flash printer firmware.
4. Click on the Browse button and choose the .hex file. The file name with the firmware for the MK3S is `MK3S_MK3S+-E3DREVO_FW_x.xx.x_MULTILANG` where the x is the firmware version number.
5. Make sure the Serial Port field displays printer's name (Original Prusa i3 MK3) and has a COM port assigned (ex. `COM4`). Click Flash! and let the procedure complete. Progress is indicated on both the Firmware flasher and the printer's display.

> [!NOTE]
> If the Serial port field displays anything but the name of the printer (ex. `usbmodemfa141` or `/dev/tty.usbmodem`) try another USB port, reinstall the driver or restart the computer.

### Post-Install
1. Reset Prusa for Revo`LCD-menu > Print from SD Card > Maintenance > reset_revo.gcode`
2. Run a PID calibration`LCD-menu > Calibration > PID calibration`
3. Run a Thermal Model Calibration`LCD-menu > Calibration > Thermal Model cal.`
4. Run a First Layer Calibration`LCD-menu > Calibration > First Layer Calibration`

**Comment by dpgoverride** [comment]
This is not my fix but a fix that I found by YouTube user @n.neves1776 as a random comment on a video over this issue by @nickyDoes.

So far I have passed all calibrations successfully without a thermal error or abort. I still need to do a test print but I am waiting on a new fan to arrive in the mail.

"Connect your printer to your PC via Pronterface. In the interface start sending the following commands:
1.	Send M503 to show current settings
2.	Send M310 S0 to temporary disable Thermal Management for PID cal
3.	Got to LCD -> Calibration -> PID cal.
4.	M310 S1 to enable Thermal Management
5.	Wait until the printer is cold before you continue.
6.	Send `G28 W" to home all without mesh bed level
7.	Send G1 X125 Y105 Z1 to move the hotend to the middle of the bed
8.	Send M155 S1 C3 to auto report temps and fans
9.	Send M310 P40 U-0.0014 V1.05 D0.15 L270 to set REVO values
10.	Send D70 S1 to enable tml debug code
11.	Send M310 A F0 to run Thermal model cal. without checking existing Thermal Management settings
12.	Send M500 to store TM values
It should all run smoothly. After, disconnect the printer and run first the Thermal Management calibration then the PID ( letting it cool between calibrations). You should be good to go after. But really, this was not a cheap kit, they know of this issue, and I hope someone in the software department was fired."

Hope it helps someone!

**Comment by VladoPortos** [comment]
@dpgoverride your solution seems to be working so far, I'm on the last step without issue. Maybe just to add that between step 11 and 12 you need to wait for it to finish :) 

I tried before full factory reset, their gcode file reset and nothing worked... this steps at least got me further, I'm off to test print next... if that fail back to old firmware I go... 

Kind of strange Prusa referring people to E3D for support... what the hell ? It was working perfectly on one firmware ( I was printing without a fail like a year ...) and it does not work on new firmware... kind of clear who caused the issue :D

**Comment by 3d-gussner** [comment]
> It was working perfectly on one firmware ( I was printing without a fail like a year ...) and it does not work on new firmware... kind of clear who caused the issue :D

Yes the Thermal Model has been introduced by Prusa and already showed that it is useful, users had bad thermistors, broken cable, etc. This prevented more damage and increased the safety of the printers.

If you think it is useless you can disable the Thermal model via gcode command. But as mentioned above even REVO users investigated after TM reported issues and found a exposed revo cable in the wire loom.

> Kind of strange Prusa referring people to E3D for support... what the hell ?

We refer to E3D as the REVO is "their" product and a 3rd Party mod for the MK3S.
When we introduced the ThermalModel we contacted E3D and agreed on that E3D will support the REVO and Prusa ONLY provides the compiled firmware files.

**Comment by VladoPortos** [comment]
@3d-gussner do you have some more info about this issue ? Cause the hotend and probe are just two type of resistors as far as I know... one for reading temperature and other to create it.... if I had been printing with the same hotend for a year, without a single issue or bad print... its probably not broken HW right ? Plus the 12 step fix above fixed it without disabling the feature.... since the new thermal model was introduced is more likely that is tailored only on the stock printer, bit too much ... even the revo firmware version had the problem... Also some people reporting that upgrading the board to 3.5s solved the issue...

**Comment by 3d-gussner** [comment]
@VladoPortos Sure

>Cause the hotend and probe are just two type of resistors as far as I know... one for reading temperature and other to create it....

With a "normal" hotend it is more or less straight forward. The heater cartridge has a resistance and heats up. Also the thermistor changes the resistance with the temperature.

The REVO is bit different. I haven't test it but according to some articles the heater core can't overheat as it changes the resistance depending on the temp. Please don't quote me here. More info you should find at E3D.

>if I had been printing with the same hotend for a year, without a single issue or bad print... its probably not broken HW right ?

There might be no issue with the electronics, but these couldn't be detected by the standard thermal protection.
The Thermal Model is way more faster than the thermal protection and can also detect "half" broken wires.
As an example one user had issues while printing where TM reported an issue. It turned out that there the insulation of wires were exposed and at specific location/orientation they caused temporary issue. The user investigated why TM was reporting and found the issue, replaced the part and has back a safe printer.

  
 >Plus the 12 step fix above fixed it without disabling the feature.... since the new thermal model was introduced is more likely that is tailored only on the stock printer, bit too much ... even the revo firmware version had the problem...

The REVO group was very small see https://github.com/prusa3d/Prusa-Firmware/issues/4105 and so we had limited information and data. Prusa tests the heaters and thermistors and we could use tons of data to adjust the TM model for the E3Dv6 hotends. The TM on stock MK3S works well.

The 12 Steps provided by the community are fine tuning the specific used REVO so the TM works better with that data.
All default PID values, for the E3Dv6 or REVO, are an average of many and as some may be "just" out of the range and the temp will oscillate to get to a stable target temp.

You can try to plot the temp curve with the default PID values and then with calibrated PID values. You will see that the difference.

As TM uses PID and other values to simulate the expected temp curve. A "bad" PID will trigger events.

>Also some people reporting that upgrading the board to 3.5s solved the issue...

Yes that is true as the Buddy firmware doesn't have the TM implemented yet. 🤞 It will be added at some point.
In my opinion the TM has more advantages than disadvantages. You are able to find even (in the beginning) minor issues which could get worse over time and so make the printer "safer".


Hope that helps.

**Comment by DiomedesDominguez** [comment]
I was getting to much TM errors after each PID tunning
```
Recv: TM: error |1.211755|>1.200000
Recv: TM: error |1.208423|>1.200000
Recv: TM: error |1.205205|>1.200000
Recv: TM: error |1.251226|>1.200000
Recv: TM: error |1.239462|>1.200000
```
I followed the Post-install steps and now everything is fine. Just to share my setup, I'm using stock E3D V6 hotend with a brand new NTC 3950 100k Thermistor, 24V 70W Ceramic Cartridge Heater and a 0.5mm bozzle nozzle. I tried to recompile but at the end only following the steps was just necessary, I took the E3D_Revo 60w HF firmware and do everything.

**Comment by 3d-gussner** [comment]
@DiomedesDominguez As you have a 70W ceramic heater I would suggest that you use the stock firmware 
and read https://github.com/prusa3d/Prusa-Firmware/blob/MK3/Firmware/variants/README.md

I  guess the ceramic heater behaves more like the original E3D heater and less like the REVO.

You can also try `M310` to change the values see https://reprap.org/wiki/G-code#M310:_Temperature_model_settings

Please be careful with the 70W heater in stock E3D aluminum block it would be better if you use a copper block (not sure if it is still sold).

What is your goal to use the 70W heater? Faster heat-up time or higher temperatures?
